### PR TITLE
Fixes #691 - remove com.ibm.icu from dependencies

### LINKED ISSUE #691: Remove all references to com.ibm.icu.text in jdt.ui and replace with java.text
Remove dependency on com.ibm.icu as it is not necessary.

### PR DESCRIPTION
**Pull request by robstryker**
## What it does

Removes a dependency on com.ibm.icu and replaces it with standard java.text APIs. 

## How to test

While most of the modified lines were fine, there was an issue with the dependency on Bidi in `JavaSourceViewer`, because the `getLevels()` function was not present. 

Looked into the code, and I believe this solution should work, but I'm not really sure how to test this. 

## Author checklist

- [x] I have thoroughly tested my changes - NO - I am not sure how to test the JavaSourceViewer / what to look for in behavior changes. 
- [X] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [X] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by robstryker** [comment]
I've triggered the bidi code path by modifying java code templates to display the JavaSourceViewer, and then pasting hebrew or arabic text inside. 

I haven't seen any exceptions yet, so I'm going to assume this works as expected.

**Comment by iloveeclipse** [comment]
@robstryker : could you please fix commit message?

Ideally first sentence summarizes the change.
After a new line you could then add a link to the issue (full url please)
Something like this:
```
Remove com.ibm.icu from dependencies

Fixes https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/691
```

**Comment by iloveeclipse** [comment]
> @robstryker : could you please fix commit message?

I guess you've missed [my comment above](https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/692#issuecomment-1664299870)?

**Comment by robstryker** [comment]
Sorry I'll get to it today.  I got distracted and Monday was a holiday in Canada (not that I'm in Canada, but the quiet day allowed me to deep-dive something i was working on without disruption)

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,9 @@ Snap.*
 # maven
 /*/target/
 .DS_Store
+.mvn/
```
[the remark this answers is not in the document]
Comment (jjohnstn) [reply .gitignore]: Done

**Review by jjohnstn — APPROVED** [review]
Looks good.

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgrunber) [reply .gitignore]: I couldn't remember why I had the exact same files locally, so I assumed it was some part of the build producing them. Then I remembered. I think Rob was having some difficulties building eclipse.jdt.ui and I tried reproducing it with his workflow and then probably forgot about it later.

**Pull request merged**